mlibrary/lib.umich.edu
# Website 97 allow two names for leadership on department template

# Overview
[Preservation Services](https://lib.umich.edu/about-us/our-divisions-and-departments/collections/preservation-services) is changing to have two leaders for the 'Leadership' value.

The `src/templates/department.js` file is the only modified file for this change.

I rewrote the code to work with any number of leaders and have the following rules in place for determining punctuation:

1. If there's one leader, simply output the leader's name as a link as it has been done previously.
![image](https://github.com/mlibrary/lib.umich.edu/assets/6354268/d0e0ff3d-c66a-44b0-aa9f-14edb3e99dba)
2. If there are two leaders, group them with the word 'and' between their names.
![image](https://github.com/mlibrary/lib.umich.edu/assets/6354268/757de590-919e-4aca-811b-8437f5045155)
3. If there are more than two leaders, add commas between the names (plus Oxford comma) and add the word 'and' before the last list item.
![image](https://github.com/mlibrary/lib.umich.edu/assets/6354268/39a2045f-422c-4ffc-87f2-5522ca61b43c)

Though I find it unlikely that there will be more than two leaders, I feel like this makes the code more robust and future-proof. Additionally, this logic can be used in other areas where the possibility of more than two values is higher.

This does assume that the `relationships.field_department_head` value is an array. Since it was previously an object there was a bit of difficulty in getting this all to sync up. I did create a check on the `relationships.field_department_head` to determine the `typeOf` and output the proper react fragment depending on its type, but the `relationships.field_department_head` should always be an array after this update.

> This pull request resolves [WEBSITE-97- Allow two names for "Leadership" on Department template](https://mlit.atlassian.net/jira/software/projects/WEBSITE/boards/27?selectedIssue=WEBSITE-97).

## Anything else?
There may be a need to check the `relationships.field_department_head` value prior to running `map()` function, but this is very unlikely.

One thing I learned from this is any changes to the Drupal data do not percolate to GraphQL on dev unless you sync production data to dev data. I'm a bit ignorant on the back-end / Drupal data so this is my best interpretation.

## Testing
The multiple values only exist on dev until the prod value gets updated later today. To test against dev use the following:
`DRUPAL_URL=https://cms.dev.lib.umich.edu/ npm run start`

- Make sure the PR is consistent in these browsers:
- Run accessibility tests:
  - [x] NVDA (screen reader test works well and reads the multiple links without stopping after each link
